source/1_engine/rex_engine/src/string/stringid.cpp

```diff
@@ -0,0 +1,137 @@
+#include "rex_engine/string/stringid.h"
+
+#include "rex_engine/string/stringpool.h"
+#include "rex_std/memory.h"
+#include "rex_std/vector.h"
+
+namespace rex
+{
+  namespace internal
+  {
+    //-------------------------------------------------------------------------
+    StringEntryID make(rsl::string_view characters)
+    {
+      if(characters.empty())
```

Comment:
should we warn about this?

Reply:
Not sure, the empty one is capture with the default ctor from StringID, you only enter the ::make function when you try and store a string. so perhaps I leave this up to you.

Reply:
the default one doesn't go into make so I'd say we should error or at least warn here as this means a StringID is used on an empty string (StringID are used on data, not sure why we need an empty one there). If we discover a case where a empty string used by a string id makes sense, we can revisit this

Reply:
better to put constrains in first and release them later than the other way around